### PR TITLE
Add checkpoint metadata fields

### PR DESCRIPTION
**Pull request by nfcampos**
- source: input, update or loop
- step: int
- make step counter continue from previous last step

### CODE REVIEW

File: langgraph/checkpoint/aiosqlite.py [hunk 1]
```diff
@@ -88,7 +89,7 @@ async def aget_tuple(self, config: RunnableConfig) -> Optional[CheckpointTuple]:
                     return CheckpointTuple(
                         config,
                         self.serde.loads(value[0]),
-                        self.serde.loads(value[2]) if value[2] is not None else None,
+                        self.serde.loads(value[2]) if value[2] is not None else {},
```
Comment (hinthornw) [inline langgraph/checkpoint/aiosqlite.py:92]: why is this needed?

Comment (nfcampos) [reply langgraph/checkpoint/aiosqlite.py:92]: olde checkpoints don't have metadata
